### DISCUSSION TITLE
How do you specify what version vcenter & hosts are running with vcsim?

### DISCUSSION THREAD

**Discussion by ljmitch** [vmware/govmomi#3081] [Community Help] [answered]
Hi, 

I can see the -api-version option but when I use it, the vCenter server and ESXi hosts still show as 6.5.

Should they show as v7.x or v8.x hosts or is the API actually running the version I think I've specified, and that host version means nothing?

It starts fine and doesn't give me an error saying I've put anything wrong in so it may just be how I'm typing it.

vcsim -api-version 8.0          ...for example, is there anything wrong with how I'm launching it there?

Also, how do you run the pre-created docker image specifying the API version out of interest? The standard command below works fine but if someone can tell me where I should add -api-version to the docker command there that would be great!

docker run -p 8989:8989 vmware/vcsim:latest

**Comment by dougm** [comment, marked as answer]
The `-api-version` flag only sets `ServiceContent.About.ApiVersion` here:
https://github.com/vmware/govmomi/blob/c2cf82e16a6889ea41d53bd4f81514e15a0780bb/vcsim/main.go#L79

We could extend it set the version in other About fields, and ESX hosts - feel free to open an issue and include any use case(s).

Another option that'll record the version fields from a live vCenter: https://github.com/vmware/govmomi/wiki/vcsim-features#record-and-playback

For docker, you can just add any args after the image:
```console
% docker run --rm -p 8989:8989 vmware/vcsim:latest -l :8989 -api-version 8.0
% docker run --rm vmware/govc:latest /govc about -k -u 172.17.0.2:8989
FullName:     VMware vCenter Server 6.5.0 build-5973321 (govmomi simulator)
...
API version:  8.0
```

**Comment by siddharth178** [comment]
If I want to simulate vCenter version7 or beyond that using vcsim, is it supported?
How setting api-version to different version affects vcsim's behaviour?
